File: pastor.py
#!/usr/bin/python

import os
import hashlib
import getpass
import pyperclip
from pbkdf2 import crypt

password_length = 16

base_phrase = getpass.getpass("Enter base phrase: ")

sanity_check = hashlib.sha256()
sanity_check.update(base_phrase)

print sum([ord(x) for x in sanity_check.digest()])

valid_characters = 'abcdefghijklmnopqrstuvwxyz0123456789.~!@#$%^&*()_+'

while (True):
    door_id = raw_input("Enter door id: ")
    if (door_id == ''):
        break

    key = crypt(door_id, base_phrase, 1000)
    key_data = hashlib.sha256()
    key_data.update(key)

    password = ''.join([valid_characters[ord(x) % len(valid_characters)] for x in key_data.digest()][:password_length])
    pyperclip.setcb(password)
    print pyperclip.getcb()

os.system('clear')

File: pastor.go
package main

import (
	"crypto/sha256"
	"fmt"
	"strings"
)

var passwordLength = 12 - 5
var basePhrase string
var validChars = "abcdefghijklmnopqrstuvwxyz0123456789"
var sanityCheck string

func main() {
	// reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter base phrase: ")
	// basePhrase, _ := reader.ReadString('\n')
	basePhrase := "crap\n"
	basePhrase = strings.TrimSpace(basePhrase)
	fmt.Print(basePhrase)

	data := sha256.Sum256([]byte(basePhrase))

	fmt.Printf("%s", data)

	// for true {
	// 	fmt.Print("Enter door id: ")
	// 	doorID, _ := reader.ReadString('\n')
	// 	doorID = strings.TrimSpace(doorID)
	// 	if doorID == "" {
	// 		os.Exit(0)
	// 	}
	// 	fmt.Print(doorID)
	// }
}
